src/io/flutter/utils/EvalOnDartLibrary.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.utils;
+
+import com.google.common.collect.Iterables;
+import com.google.gson.JsonObject;
+import com.intellij.openapi.Disposable;
+import com.intellij.util.Alarm;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.IsolatesInfo;
+import io.flutter.run.FlutterDebugProcess;
+import org.dartlang.vm.service.VmService;
+import org.dartlang.vm.service.consumer.EvaluateConsumer;
+import org.dartlang.vm.service.consumer.GetIsolateConsumer;
+import org.dartlang.vm.service.consumer.GetObjectConsumer;
+import org.dartlang.vm.service.element.*;
+
+import java.lang.System;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Invoke methods on a Dart library using the observatory protocol.
+ */
+public class EvalOnDartLibrary implements Disposable {
+  private final IsolatesInfo.IsolateInfo isolateInfo;
+  private final VmService vmService;
+  private Isolate isolate;
+  private final String libraryName;
+  final CompletableFuture<LibraryRef> libraryRef;
+  private final Alarm myRequestsScheduler;
+
+  public EvalOnDartLibrary(String libraryName, FlutterDebugProcess debugProcess, VmService vmService) {
+    this.vmService = vmService;
+    final Collection<IsolatesInfo.IsolateInfo> isolates = debugProcess.getIsolateInfos();
+    assert(isolates.size() == 1);
+    isolateInfo = Iterables.get(isolates, 0);
+    this.libraryName = libraryName;
+    this.myRequestsScheduler = new Alarm(Alarm.ThreadToUse.POOLED_THREAD, this);
+    libraryRef = new CompletableFuture<>();
+    initialize();
+  }
+
+  public void dispose() {
+    myRequestsScheduler.dispose();
+    // TODO(jacobr): complete all pending futures as cancelled?
+  }
+
+  /**
+   * Workaround until the version of the VmService 'eval' method supports
+   * the scope argument.
+   */
+  private void callVmServiceRequest(VmService vmService, String methodName, JsonObject params, EvaluateConsumer consumer) {
+    try {
+      final Class<?> c = Class.forName("org.dartlang.vm.service.VmServiceBase");
+
+      final Method[] allMethods = c.getDeclaredMethods();
```

Comment:
There is a `com.intellij.util.ReflectionUtil` class which can reduce a bit of the work here (used for example by `FlutterReloadManager`)

Reply:
That is handy. Done.